cbor.js

```diff
@@ -344,13 +476,15 @@ function decode(data, tagger, simpleValue) {
         }
         return readArrayBuffer(length);
       case 3:
-        var utf16data = [];
+        var newStr = "";
         if (length < 0) {
-          while ((length = readIndefiniteStringLength(majorType)) >= 0)
-            appendUtf16Data(utf16data, length);
-        } else
-          appendUtf16Data(utf16data, length);
-        return String.fromCharCode.apply(null, utf16data);
+          while ((length = readIndefiniteStringLength(majorType)) >= 0) {
+            newStr = appendUtf16Data(newStr, length);
+          }
+        } else {
+          newStr = appendUtf16Data(newStr, length);
+        }
+        return newStr;
```

Comment:
just created this one (revised several times to ensure accuracy on large and small strings, and relevance to scenario) - https://jsfiddle.net/dyj47zdq/18/ - array push and String.fromCharCode are faster than string append in browsers.
change was originally based on benchmarks of _just appending strings_ (not created by String.fromCharCode) and where array join was used instead of fromCharCode.
good to revert this change.